master/internal/api_tasks.go

```diff
@@ -300,6 +300,43 @@ func (a *apiServer) TaskLogs(
 	})
 }
 
+func (a *apiServer) PostTaskLogs(
+	ctx context.Context, req *apiv1.PostTaskLogsRequest,
+) (*apiv1.PostTaskLogsResponse, error) {
+	if len(req.Logs) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "len logs must be greater than 0")
+	}
+	taskID := req.Logs[0].TaskId
+
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(taskID),
+		expauth.AuthZProvider.Get().CanEditExperiment); err != nil {
+		return nil, err
+	}
+
+	logs := make([]*model.TaskLog, len(req.Logs))
+	for i := range req.Logs {
+		if req.Logs[i].Id != nil {
```

Comment:
question (non-blocking): Why are we checking this field? And why is it required to be nil? Is it prevent confusion?

Reply:
We ignore this field in insert task logs since it is autogenerated at the db.

https://github.com/determined-ai/determined/blob/588d817d70bbfdc4860c6d1262be061cdce9761c/master/internal/db/postgres_tasks.go#L362C1-L362C22

I just made it required to be nil so people aren't setting something then wondering why it has no effect.

I guess the field is technically unused but we should probably switch some endpoints to use the task logs proto object for consistency eventually